### PR TITLE
Remove Connector and Instance legacy client APIs

### PR DESCRIPTION
**Pull request by ctubbsii**

### CODE REVIEW

**Comment by ctubbsii** [comment]
This was a much smaller set of changes than I expected.

File: core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
  *
  * <p>
  * If migrating code from Connector to AccumuloClient an important difference to consider is that
- * AccumuloClient is closable and Connector is not. Connector uses static resources and therefore
+ * AccumuloClient is closable and Connector was not. Connector uses static resources and therefore
  * creating them is cheap. AccumuloClient attempts to clean up resources on close, so constantly
```
Comment (EdColeman) [inline core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java:56]: The Connector used static resources and was cheaper to create that the AccumuloClient.  However, ...

File: core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java [hunk 2]
```diff
@@ -52,7 +52,7 @@
...
  * creating them could perform worse than Connector. Therefore, it would be better to create an
```
Comment (EdColeman) [inline core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java:57]: Therefore, it is better to ...

**Review by EdColeman — APPROVED** [review]
LGTM - just some minor doc change suggestions

File: core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java (hunk 1 above)
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: Addressed in 49dd2789c4

File: core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java (hunk 2 above)
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: Addressed in 49dd2789c4

**Comment by ctubbsii** [comment]
> LGTM - just some minor doc change suggestions

Thanks. I think I updated the wording to be better than what was there before. Will wait on full ITs before merging, though.

**Comment by ctubbsii** [comment]
All ITs passed

**Pull request merged**